src/main/java/org/jabref/gui/preferences/PreviewTabViewModel.java

```diff
@@ -0,0 +1,188 @@
+import org.jabref.preferences.PreviewPreferences;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PreviewTabViewModel implements PreferenceTabViewModel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PreviewTabViewModel.class);
+
+    private final ListProperty<PreviewLayout> availableListProperty = new SimpleListProperty<>();
+    private final ListProperty<PreviewLayout> selectedAvailableItemsProperty = new SimpleListProperty<>();
+    private final ListProperty<PreviewLayout> chosenListProperty = new SimpleListProperty<>();
+    private final ListProperty<PreviewLayout> selectedChosenItemsProperty = new SimpleListProperty<>();
+
+    private final StringProperty previewTextProperty = new SimpleStringProperty("");
+
+    private final DialogService dialogService;
+    private final JabRefPreferences preferences;
+    private final TaskExecutor taskExecutor;
+
+    public PreviewTabViewModel(DialogService dialogService, JabRefPreferences preferences, TaskExecutor taskExecutor) {
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        this.taskExecutor = taskExecutor;
+
+        setValues();
+    }
+
+    public void setValues() {
+        PreviewPreferences previewPreferences = preferences.getPreviewPreferences();
+        ObservableList<PreviewLayout> availableLayouts = FXCollections.observableArrayList();
+
+        chosenListProperty.setValue(FXCollections.observableArrayList(previewPreferences.getPreviewCycle()));
+
+        availableLayouts.clear();
+        if (chosenListProperty.stream().noneMatch(layout -> layout instanceof TextBasedPreviewLayout)) {
+            availableLayouts.add(previewPreferences.getTextBasedPreviewLayout());
+        }
+
+        BackgroundTask.wrap(CitationStyle::discoverCitationStyles)
+                      .onSuccess(value -> value.stream()
+                                               .map(CitationStylePreviewLayout::new)
+                                               .filter(style -> !chosenListProperty.contains(style))
+                                               .sorted(Comparator.comparing(PreviewLayout::getName))
+                                               .forEach(availableLayouts::add))
+                      .onFailure(ex -> {
+                          LOGGER.error("something went wrong while adding the discovered CitationStyles to the list ", ex);
+                          dialogService.showErrorDialogAndWait(Localization.lang("Error adding discovered CitationStyles"), ex);
+                      })
+                      .executeWith(taskExecutor);
+
+        availableListProperty.setValue(availableLayouts);
+
+        previewTextProperty.setValue(previewPreferences.getPreviewStyle().replace("__NEWLINE__", "\n"));
+    }
+
+    @Override
+    public void storeSettings() {
+        PreviewPreferences previewPreferences = Globals.prefs.getPreviewPreferences();
```

Comment:
preferences.getPreviewPreferences() ? as you already have a JabRef prefs object

Reply:
Thanks, it's been late. 😄